CONTRIBUTING.md

```diff
@@ -37,27 +37,29 @@ to avoid duplication of effort.
 The process of submitting a Pull Request is relatively simple, following a 
 [rebase workflow](http://kensheedlo.com/essays/why-you-should-use-a-rebase-workflow/).  Please use the
 following pattern:
+
 1. Create a new branch for your fix/feature
 1. Make your changes
-1. Rebase onto the upstream
-1. Run the test suite
-1. Squash your commits
+1. Run the test suite *(if failed, goto 2)*
+1. Squash your commits *(re-run test suite)*
+1. Rebase onto the upstream *(re-run test suite)*
```

Comment:
The only problem comes when you try to rebase onto upstream *before* squashing, and it gives you a mega-merge-conflict because it wants to merge *all* commits, rather than your atomic change. It provides a better workflow for the user, and I'd hate for a change to be sidelined because of a 2-hour merge conflict exercise, while git can do it just fine when there's only one commit to rebase.